ocw_data_parser/course_downloader.py

```diff
@@ -54,10 +55,10 @@ def download_courses(self):
                             with open(dest_filename, "wb+") as f:
                                 s3_client.download_fileobj(
                                     self.s3_bucket_name, obj["Key"], f)
-                        courses.pop(courses.index(course_id))
+                                if course_id not in downloaded_courses:
```

Comment:
Is the if statement needed?

Reply:
if it's not there, it will append the course ID to the list for every numbered json file in the bucket and unnecessarily create a huge list.